### PR TITLE
clean up, response validation

### LINKED ISSUE #14: Works against HTTP server
This plugin return OK when checking the port of an HTTP server (in tcp mode), just because, regardless of the query, the HTTP server answers something. That would be nice if this plugin checked whether the data returned fitted the openvpn procol.
A possible use case is to monitor that sslh is working fine.

### PR DESCRIPTION
**Pull request by andiwand**
should fix #14

### CODE REVIEW

**Comment by liquidat** [comment]
@andiwand As usual, thanks for this impressive PR - how about we move the code to a repository where you get full rights as well? :)

One thing though before I merge this: currently if you check a VPN server without any further details like a client key, the check fails because the return function does try to return a value which was not set:
```
./check_openvpn boran.bayz.de 
Traceback (most recent call last):
  File "./check_openvpn", line 257, in <module>
    code = main()
  File "./check_openvpn", line 254, in main
    return check(args.host, args.port, args.tcp, args.timeout, digest, digest_size, client_key, server_key, args.retrycount, args.validate)
UnboundLocalError: local variable 'client_key' referenced before assignment
```

File: check_openvpn [hunk 1]
```diff
@@ -182,15 +236,22 @@ def main(argv=None):
...
+        index_start = MAX_CIPHER_KEY_LENGTH
+        index_end = index_start + MAX_HMAC_KEY_LENGTH
+        server_key = key[index_start:index_end]
+        if args.tls_auth_inverse: client_key, server_key = server_key, client_key
+        # reduce key size to required size
+        client_key = client_key[:digest_size]
+        server_key = server_key[:digest_size]
 
-    return checkserver(args.host, args.port, args.tcp, args.timeout, key, digest, args.retrycount)
+    return check(args.host, args.port, args.tcp, args.timeout, digest, digest_size, client_key, server_key, args.retrycount, args.validate)
```
Comment (liquidat) [inline check_openvpn]: This requires that client_key and others are set, but that is not always the case.

**Comment by andiwand** [comment]
@liquidat thanks! fixed that and tested against https server.

**Comment by liquidat** [comment]
@andiwand 
```
$ ./check_openvpn -p 443 -t boran.bayz.de
CRIT: Invalid response
$ ./check_openvpn -p 443 -t tuzak.bayz.de                                                                                                                                                CRIT: Invalid response
```
tuzak.bayz.de on port 443 is an http server - there the critical response is correct.
boran.bayz.de on port 443 is a VPN server (sorry, I should have mentioned that) ;)

So the current critical response seems to be wrong, the server is working fine for me. Not sure where the error is though... :-/

And regarding code moving: you are right that this place is well known already... but if you ever feel like moving the code, just let me know, I would help with it. Until then I hunt you with tests and bugs ;)

**Comment by andiwand** [comment]
@liquidat if you could test it again... i hope for the last time :P

**Comment by liquidat** [comment]
Tests run perfectly fine! I have not tested the key feature yet but trust you that it works as expected.

Thanks again! I will merge it. How about we add you to the list of contributors in the source code file?

**Comment by andiwand** [comment]
Yeah i tested the key feature! And i would appreciate that : )

**Pull request merged**